week3/intro_to_costing.ipynb

```diff
@@ -0,0 +1,779 @@
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Solver Status: ok, Termination Condition: optimal\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Solve model\n",
+    "results = solver.solve(m)\n",
+    "assert_optimal_termination(results)\n",
+    "print(\n",
+    "    f\"Solver Status: {results.solver.status}, Termination Condition: {results.solver.termination_condition}\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "57ef7c1b-8dc5-4898-ac87-85dad0360c93",
+   "metadata": {},
+   "source": [
+    "## Try It Yourself\n",
+    "\n",
+    "Add a `UnitModelCostingBlock` for a high-pressure pump and change the cost from 1.908 USD/W to 1 USD/W given the following information. Afterwards, re-solve the model.\n",
+    "<p align=\"center\">\n",
+    "<img src='graphics/pump_costing_doc.png' alt=\"Pump Costing Assumptions\" width=\"1000\" height=800>\n",
+    "</p>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "e7954af0-f3c7-4cb2-ae91-c2b3d5d1c77e",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Block fs.costing.high_pressure_pump\n",
+      "\n",
+      "  Variables:\n",
+      "    cost : High pressure pump cost\n",
+      "        Size=1, Index=None, Units=USD_2018/W\n",
+      "        Key  : Lower : Value : Upper : Fixed : Stale : Domain\n",
+      "        None :     0 : 1.908 :  None :  True : False :  Reals\n",
+      "\n",
+      "  Objectives:\n",
+      "    None\n",
+      "\n",
+      "  Constraints:\n",
+      "    None\n",
+      "Solver Status: ok, Termination Condition: optimal\n",
+      "Block fs.costing.high_pressure_pump\n",
+      "\n",
+      "  Variables:\n",
+      "    cost : High pressure pump cost\n",
+      "        Size=1, Index=None, Units=USD_2018/W\n",
+      "        Key  : Lower : Value : Upper : Fixed : Stale : Domain\n",
+      "        None :     0 :     1 :  None :  True :  True :  Reals\n",
+      "\n",
+      "  Objectives:\n",
+      "    None\n",
+      "\n",
+      "  Constraints:\n",
+      "    None\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Add pump UnitModelCostingBlock\n",
+    "m.fs.pump.costing = UnitModelCostingBlock(\n",
+    "    flowsheet_costing_block=m.fs.costing,\n",
+    "    costing_method=cost_high_pressure_pump,\n",
+    ")\n",
+    "\n",
+    "# Print the cost model for the pump before updating\n",
+    "m.fs.costing.high_pressure_pump.display()\n",
+    "\n",
+    "# Update pump cost parameter\n",
+    "m.fs.costing.high_pressure_pump.cost.fix(1)\n",
+    "\n",
+    "results = solver.solve(m)\n",
+    "assert_optimal_termination(results)\n",
+    "print(\n",
+    "    f\"Solver Status: {results.solver.status}, Termination Condition: {results.solver.termination_condition}\"\n",
+    ")\n",
+    "\n",
+    "# Print the cost model for the pump to verify changes\n",
+    "m.fs.costing.high_pressure_pump.display()"
+   ]
```

Comment:
I've added the solution for the TIY just so that we have a record of what the solution might look like. This should be deleted before merging. We'll also probably need to give better instructions/hints for them to be able to do this successfully